rfcs/20210129-optimize-pd-creation/README.md

```diff
@@ -0,0 +1,262 @@
+# Optimization of Primitive Descriptor Creation
+
+## Motivation
+
+There are three majors steps to create a oneDNN primitive:
+1. Create operation descriptor
+2. Create primitive descriptor
+3. Create primitive
+
+In frameworks, all the three steps are performed every time when a user executes
+an operation. The last two steps are expensive hence they can significantly
+affect performance of the operation.
+
+oneDNN recently introduced a cache for primitives to significantly reduce primitive
+creation time - the most expensive step. The second most expensive step is creating
+the primitive descriptor. The cost may very from several microsecond to hundreds
+of microseconds depending on number of the primitive implementations and time of
+primitive descriptor initialization.
+Such overhead is significant because the execution time can also fall in this range.
+This is the motivation to start optimizing the primitive descriptor creation step.
+
+
+## Proposal
+
+Introducing one more cache for primitive descriptors doesn't look appealing for
```

Comment:
Do I understand it right that by `one more cache` you mostly mean a separate and dedicated key for primitive descriptors? Is it because a dedicated to pd key would require a dedicated search and a separate space to store them?

Reply:
One more cache is about one more cache entity in the library and all the consequences of that, which include increased memory consumption, new API, additional maintenance cost, etc...

The key can be generalized, this is the main goal of this RFC.